src/uct/tcp/tcp.h

```diff
@@ -16,6 +16,41 @@
 /** How many events to wait for in epoll_wait */
 #define UCT_TCP_MAX_EVENTS        32
 
+#define UCT_TCP_AM_SHORT_PACK_DATA(_pack_f, _target_buf, _target_length, \
+                                   _am_payload, _payload_length, _am_header) \
+    do { \
+        _pack_f(_target_buf, &(_am_header), sizeof(_am_header)); \
+        _pack_f((uint8_t*)(_target_buf) + sizeof(_am_header), \
+                _am_payload, _payload_length); \
+        _target_length = sizeof(_am_header) + _payload_length; \
+    } while (0)
+
+#define UCT_TCP_AM_BCOPY_PACK_DATA(_pack_f, _target_buf, _target_length, \
+                                   _am_arg, ...) \
+    do { \
```

Comment:
minor: do you need do/while around single operation?

Reply:
👍 there is no need -- fixed